### PR TITLE
Allow Batch Updates

### PR DESCRIPTION
**Pull request by dheerajodha**

### CODE REVIEW

File: src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java [hunk 1]
```diff
@@ -43,4 +46,16 @@ public Plugin saveOrUpdate(Plugin plugin) {
...
+        for (int i = 0; i < pluginList.size(); i++) {
+            Optional<Plugin> pluginFromDatabase = pluginRepository.findByName(pluginList.get(i).getName());
+
+            if (pluginFromDatabase.isPresent()) {
+                pluginList.set(i, pluginFromDatabase.get().setScm(pluginList.get(i).getScm()).setReleaseTimestamp(pluginList.get(i).getReleaseTimestamp()));
+            }
+        }
+
+        return pluginRepository.saveAll(pluginList);
+    }
```
Comment (alecharp) [inline src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java:60]: Here, you are transforming the same list that you are iterating on. This cannot be correct and you should have concurrent access issues running this code.

However, you could leverage `Stream` API to find a plugin in DB, if it exists, then set the fields you want and finally collect that into a list that you send to the `Repository#saveAll` method.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -4,6 +4,10 @@ spring.datasource.password=${POSTGRES_PASSWORD}
 spring.datasource.driverClassName=org.postgresql.Driver
 spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation=true
 spring.jpa.properties.hibernate.dialect= org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.properties.hibernate.generate_statistics=true
+spring.jpa.properties.hibernate.jdbc.batch_size=${BATCH_SIZE}
+spring.jpa.properties.hibernate.order_inserts=true
+spring.jpa.properties.hibernate.order_updates=true
```
Comment (alecharp) [inline src/main/resources/application.properties:10]: This file was removed and replaced by `application.yml`. I believe that you didn't update your `main` branch with the one on this repository. 
Please be careful with this. When a PR is merged, you need to `git pull --rebase` your clone.

**Review by alecharp — CHANGES_REQUESTED** [review]

**Comment by dheerajodha** [comment]
Sorry @alecharp, I had to close this PR while rebasing. Because my local branch didn't have the 2 recently-merged PRs in it, and when I tried to rebase, those 2 PRs were sitting on top of recent commits related to this PR. So, I did: `git reset --hard` 3fc8dd3656bdf6384c5d633cadadac33d5af5e41, followed by `git pull upstream main`, then added my changes related to this PR, and committed (to maintain the correct order of the commits). And had to delete this branch due to the change.

File: src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java (hunk 1 above)
Comment (dheerajodha) [reply src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java]: Thanks for pointing out the mistake related to concurrent access, I totally forgot about it.
I'm now using `Stream`  API in the new PR, please find it here: https://github.com/jenkins-infra/plugin-health-scoring/pull/26/files

File: src/main/resources/application.properties (hunk 2 above)
Comment (dheerajodha) [reply src/main/resources/application.properties]: Thanks, I will keep this in mind :+1:

**Pull request closed without merging**